frontend/admin/src/js/components/pool/EditPool/EditPool.test.tsx

```diff
@@ -84,7 +84,10 @@ describe("Edit Pool tests", () => {
     // render story and click the button to open the modal
     await act(async () => {
       render(<DraftAdvertisement {...props} />);
-      fireEvent.click(screen.getByRole("button", { name: /publish/i }));
+    });
+
+    await act(async () => {
```

Comment:
> **Note**
> I was having trouble getting this test to pass consistently here and moving the dialog open to it's own act worked. I believe it is because we are rendering a new form in the dialog which must be wrapped in `act()` due to how `react-hook-form` renders work.

Reply:
act should not be needed around any fireEvent, so if it's helping, it's because it's just slowing things down and avoiding a race condition that still exists (and might crop up sporadically)

If you feel skeptical of whether act should be avoided, i can dig up the issue from the test-library author where he said this unambiguously